src/rotation_types.jl

```diff
@@ -0,0 +1,192 @@
+
+
+# a default element type
+macro DefaultElType(); Float64; end  # need this to be a macro for type stability in type_methods.jl promote_type_sp functions
+
+# some trait style stuff
+numel{T}(::Type{T}) = length(fieldnames(T))    # function to get the number of elements in each parameterization
```

Comment:
yeah; I need something different to length for Quaternions since they have length 1 to match being a subtype of number